.changeset/tiny-maps-study.md

```diff
@@ -0,0 +1,171 @@
+---
+'@makeswift/runtime': minor
+---
+
+BREAKING: Reworks how the Makeswift builder displays your site by leveraging Next.js' Preview Mode!
+
+This is a _huge_ change and makes integrating Makeswift into your Next.js app a lot simpler. We've
+deprecated the `getStaticPaths`, `getStaticProps`, and `getServerSideProps` exports from
+`@makeswift/runtime/next` and will be removing them in the next minor release. We recommend you
+follow the migration steps below.
+
+Here's how to migrate:
+
+- Create a new file at `pages/api/makeswift/[...makeswift].js` with the following contents:
+
+  ```js
+  import { MakeswiftApiHandler } from '@makeswift/runtime/next'
+
+  export default MakeswiftApiHandler(process.env.MAKESWIFT_SITE_API_KEY)
+  ```
+
+- Update your dynamic optional catch-all route to use the new data fetching APIs,
+  `Makeswift.getPages` and `Makeswift.getPage`.
+
+  ```diff
+  import './path/to/makeswift/register-components'
+
+  -export { getStaticPaths, getStaticProps, Page as default }
+  +import { Makeswift, Page as MakeswiftPage } from '@makeswift/runtime/next'
+  +
+  +export async function getStaticPaths() {
+  +  const makeswift = new Makeswift(process.env.MAKESWIFT_SITE_API_KEY)
+  +  const pages = await makeswift.getPages()
+  +
+  +  return {
+  +    paths: pages.map((page) => ({
+  +      params: {
+  +        path: page.path.split('/').filter((segment) => segment !== ''),
+  +      },
+  +    })),
+  +    fallback: 'blocking',
+  +  }
+  +}
+  +
+  +export async function getStaticProps(ctx) {
+  +  const makeswift = new Makeswift(process.env.MAKESWIFT_SITE_API_KEY)
+  +  const path = '/' + (ctx.params?.path ?? []).join('/')
+  +  const snapshot = await makeswift.getPageSnapshot(path, {
+  +    preview: ctx.preview,
+  +  })
+  +
+  +  if (snapshot == null) return { notFound: true }
+  +
+  +  return { props: { snapshot } }
+  +}
+  +
+  +export default function Page({ snapshot }) {
+  +  return <MakeswiftPage snapshot={snapshot} />
+  +}
+  ```
+
+- Delete your Makeswift preview route. This page won't be used anymore. It's likely at
+  `pages/makeswift.js` and the diff might look something like this:
+
+  ```diff
+  -import './path/to/makeswift/register-components'
+  -
+  -export { getServerSideProps, Page as default } from '@makeswift/runtime/next'
+  ```
+
+- Go to your Makeswift site settings and update the host URL to be just your host's
+  origin. For example, change `https://www.makeswift.com/makeswift` to just
+  `https://www.makeswift.com` or `http://localhost:3000/makeswift` to just `http://localhost:3000`.
+
+If you have any questions about the migration or run into any issues, please don't hesitate to chat
+with us. [We're on Discord!](https://discord.gg/PkrUsFnMUn)
+
```

Comment:
Should we add a divider here or something more obvious to separate between two sections? At a first glance, it's kinda hard to read because I was assuming it was still part of the migration process.

I kinda see it like the first part is more of a TL;DR that anyone needs to read, and the second part is more of an optional detailed explanation.